lib/icinga/checkable-dependency.cpp

```diff
@@ -154,23 +320,77 @@ std::set<Checkable::Ptr> Checkable::GetAllChildren() const
 	return children;
 }
 
+/**
+ * Retrieve all direct and indirect children of the current Checkable.
+ *
+ * Note, this function performs a recursive call chain traversing all the children of the current Checkable
+ * up to a certain limit (256). When that limit is reached, it will log a warning message and abort the operation.
+ *
+ * @param children - The set of children to be filled with all the children of the current Checkable.
+ * @param level - The current level of recursion.
+ */
 void Checkable::GetAllChildrenInternal(std::set<Checkable::Ptr>& children, int level) const
 {
-	if (level > 32)
-		return;
+	if (level > l_MaxDependencyRecursionLevel) {
+		Log(LogWarning, "Checkable")
+			<< "Too many nested dependencies (>" << l_MaxDependencyRecursionLevel << ") for checkable '" << GetName() << "': aborting traversal.";
+		return ;
+	}
 
 	std::set<Checkable::Ptr> localChildren;
 
 	for (const Checkable::Ptr& checkable : children) {
-		std::set<Checkable::Ptr> cChildren = checkable->GetChildren();
-
-		if (!cChildren.empty()) {
+		if (auto cChildren(checkable->GetChildren()); !cChildren.empty()) {
 			GetAllChildrenInternal(cChildren, level + 1);
 			localChildren.insert(cChildren.begin(), cChildren.end());
 		}
 
-		localChildren.insert(checkable);
+		if (level != 0) { // Recursion level 0 is the initiator, so checkable is already in the set.
+			localChildren.insert(checkable);
+		}
 	}
```

Comment:
I'm not sure about these changes. They look like they don't really change anything. The first change just uses C++17 syntax and the second change only skips the insert of an element that should already be in the set, so the insert does nothing already?

If changing something here, I'd rather give the function a complete makeover: currently it seems like it would visit checkables multiple times which could lead to its runtime complexity exploding for adverse dependency graphs.

Reply:
Done!